core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -157,6 +161,62 @@
     @Nullable
     SSLSession sslSession();
 
+    /**
+     * Returns the {@link URI} of the current {@link Request}. Note that the scheme of the returned {@link URI}
+     * denotes the requested scheme, which may be different from the actual scheme. For example, if you sent
+     * a request to {@code http://foo.com/"}, you will always get {@code "http"} as a scheme even if the actual
+     * {@link SessionProtocol} is {@code "h2c"}. Similarly, if you sent a request to {@code h2c://foo.com/}, you
+     * will always get {@code "h2c"} as a scheme.
+     *
+     * <p>This method is a shortcut of {@code uri(false)}. If you want to make sure the scheme of the returned
+     * {@link URI} is always {@code "http"} or {@code "https"}, call {@code uri(true)}.</p>
+     *
+     * @see #uri(boolean)
+     */
+    default URI uri() {
+        return uri(false);
+    }
+
+    /**
+     * Returns the {@link URI} of the current {@link Request}.
+     *
+     * @param simplifyScheme If {@code true}, the scheme of the returned {@link URI} will always be
```

Comment:
I'm not sure I like it either, but this was introduced for the users who prefers using http/https over h1, h2, ... (Brave to name one :smile:) If we did not have this flag, a user will have to reconstruct the URI just to replace the scheme.

Reply:
I think that came up for brave when we noticed the scheme was h2c etc even though the user used http:// when defining the client (actually even more weird like none+h2c) since we were using established, not requested, scheme before. 

I feel as if the user requested h2:// it would be ok to use it should be used in instrumentation too since it's an advanced use case but explicit by the user. @adriancole does that make sense?

Reply:
it totally does not make sense, as no one would expect the tag http.uri to have custom schemes.

Reply:
someones when someone is asking for something obscure, it is good to do a poll and look for other telemetry systems. for example, how many expect or are able to parse successfully odd things like h2://

I think it would be a bug making machine to propagate this upstream iotw and it wouldn't take much querying to find majority would agree.

Reply:
See #2092. I'm closing this.